docs/developer/watcher-local-test.md

```diff
@@ -0,0 +1,129 @@
+# Test watcher locally in two-cluster-setup using K3D
+
+### Contents
+
+### Create KCP cluster
+Run the following command to create a local control-plane (KCP) cluster:
+```shell
+k3d cluster create kcp-local --port 9080:80@loadbalancer \
+--registry-create k3d-kyma-registry.localhost:0.0.0.0:5111 \
+--k3s-arg '--no-deploy=traefik@server:0'
+```
+
+### Install Istio and Manifest CRD
+Run the following command to install Istio and Manifest CRD on the KCP cluster:
+```shell
+kubectl config use-context k3d-kcp-local && \
+istioctl install -y && \
+kubectl apply -f https://raw.githubusercontent.com/kyma-project/module-manager/main/config/crd/bases/operator.kyma-project.io_manifests.yaml
+```
+
+### Build and push lifecycle-manager image
+Run the following commands to build and push lifecycle-manager image to local k3d registry:
+```shell
+export K3D_REG=k3d-kyma-registry.localhost:5111 && \
+make docker-build IMG=$K3D_REG/lifecycle-manager:latest && \
+make docker-push IMG=$K3D_REG/lifecycle-manager:latest
+```
+
+### Deploy lifecycle-manager on KCP:
+Run the following command to deploy lifecycle-manager on the KCP cluster:
+```shell
+kubectl config use-context k3d-kcp-local && \
```

Comment:
You can remove `use-context ` in each command. Rather switch once during the course of the documentation.